proposals/esm-integration/README.md

```diff
@@ -97,18 +97,18 @@ For WebAssembly, instantiation will include initialization for all [external typ
 - memories
 - globals
 
-Because WebAssembly does not currently have a type that can handle arbitrary JS values, JS value imports can not be handled with live bindings and would be snapshotted when they are imported. This means that in the short term, non-function imports coming from JS modules would be assigned a value of `undefined`, which will currently result in an error. 
+Because WebAssembly does not currently have a type that can handle arbitrary JS values, JS value imports can not be handled. This means that in the short term, non-function imports coming from JS modules will currently result in an error. 
 
-It may be possible to provide live bindings for JS values if an `any` type is added to WebAssembly. At this point the restriction prohibiting `undefined` imports could also be lifted. However, we would still want to throw an error for non-function imports in this case. 
+It may be possible to provide live bindings for JS values if an `anyref` type is added to WebAssembly. At this point the restriction prohibiting non-function imports could also be lifted.
 
 The reason we want to throw an error in this case is two fold:
 
-- **Reduce confusion for developers.** Since non-function imports work in JS->JS dependency chains, it is unintuitive that they would be undefined in JS->wasm dependency chains. Throwing an error makes clear what's happening.
-- **Allow for future support.** It may be possible to support non-function imports to wasm modules if an `any` type is added. We close off this avenue if we allow users to import values that can be undefined, though, as code may start depending on that behavior.
+- **Reduce confusion for developers.** WebAssembly snapshots imports while instantiation, but most exports from JS modules are still `undefined` or in TDZ at this time. Since non-function imports work in JS->JS dependency chains, it is unintuitive that they would be undefined in JS->wasm dependency chains. Throwing an error makes clear what's happening.
```

Comment:
This snapshotting time is the proposal at master, and this paragraph ("shapshots imports while instantiation") is characteristic of it, but I like the PR at #13, which changes the time to be a bit later (during execution) and allows more cases to be well-handled.